http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -224,6 +224,21 @@ public HttpExecuteResponse call() throws IOException {
                                       .build();
         }
 
+        /**
+         * {@link sun.net.www.protocol.http.HttpURLConnection#getInputStream0()} has been observed to intermittently throw
+         * {@link NullPointerException}s for reasons that still require further investigation, but are assumed to be due to a
+         * bug in the JDK. Propagating such NPEs is confusing for users and are not subject to being retried on by the default 
+         * retry policy configuration, so instead we bias towards propagating these as {@link IOException}s.
+         */
+        private static int getResponseCodeSafely(HttpURLConnection connection) throws IOException {
```

Comment:
Agree we should revisit this in the future. Can you add a `TODO` here?

Reply:
Will do.